torch/utils/_config_module.py

```diff
@@ -493,12 +493,61 @@ def codegen_config(self) -> str:
         """Convert config to Python statements that replicate current config.
         This does NOT include config settings that are at default values.
         """
+
+        # additional imports required
+        imports = set()
+
+        def get_module_name(func, add_dot):
+            module_name = func.__module__
```

Comment:
Not sure why simply returning module_name is not enough here.

Reply:
> Not sure why simply returning module_name is not enough here.

@desertfire For python native functions like `print`, the module name is `builtins`, but `builtins.print` is not a runnable command. So we need to remove the `builtins` part.

For `_warnings`, when you import the `warnings` module, Python actually imports a module called `_warnings` internally. The `warnings` module is a wrapper around the `_warnings` module that provides a more user-friendly interface.

In general, one should use the warnings module, rather than trying to access the _warnings module directly.

Reply:
I suspect `_warnings` will use the same API. Especially as debugging tool, we don't have to worry about API BC/FC here.